bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py

```diff
@@ -262,6 +267,291 @@ def predict(self):
                 if class_id == 1:
                     self._goalpost_candidates.append(c)
 
+class YoloHandlerNCS2(YoloHandler):
+    """
+    This Code is based on a code example from the Intel documentation under following licensing:
+
+    Copyright (C) 2018-2019 Intel Corporation
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+    """
+
+    def __init__(self, config, model_path):
+        # Create model file paths
+        model_xml = os.path.join(model_path, "yolo.xml")
+        model_bin = os.path.join(model_path, "yolo.bin")
+
+        # Plugin initialization
+        rospy.loginfo("Creating Inference Engine...")
+        ie = IECore()
+
+        # Reading the IR generated by the Model Optimizer (.xml and .bin files)
+        rospy.loginfo("Loading network files:\n\t{}\n\t{}".format(model_xml, model_bin))
+        self._net = IENetwork(model=model_xml, weights=model_bin)
+
+        assert len(self._net.inputs.keys()) == 1, "Sample supports only YOLO V3 based single input topologies"
+
+        # Preparing network inputs
+        rospy.loginfo("Preparing inputs")
+        self._input_blob = next(iter(self._net.inputs))
+
+        #  Defaulf batch_size is 1
+        self._net.batch_size = 1
+
+        # Read and pre-process input images
+        self._n, self._c, self._h, self._w = self._net.inputs[self._input_blob].shape
+
+        # Device type
+        device = "MYRIAD"
+
+        # Loading model to the plugin
+        rospy.loginfo("Loading model to the plugin")
+        self._exec_net = ie.load_network(network=self._net, num_requests=2, device_name=device)
+
+        # Params TODO
```

Comment:
plz dyn recnf

Reply:
This will be done for all yolo handlers after this pull request is merged. There is an issue for this.